.github/workflows/codecov.yml

```diff
@@ -28,4 +28,4 @@ jobs:
       run: |
         python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder')"
         chmod a+x builder
-        ./builder build -p ${{ env.PACKAGE_NAME }} --compiler=gcc-9 --cmake-extra=-DASSERT_LOCK_HELD=ON --coverage
+        ./builder build -p ${{ env.PACKAGE_NAME }} --compiler=gcc-12 --cmake-extra=-DASSERT_LOCK_HELD=ON --coverage
```

Comment:
why? Add some comments about this?
Do we need it to be applied to other codecov jobs?

Reply:
I have added the comment. Yes, we need to fix it everywhere where we updated the code coverage action ubuntu version.

Reply:
hmmm looks like the ci before the commit changing the compiler still work? https://github.com/awslabs/aws-c-auth/actions/runs/12757805423/job/35558824650

Reply:
If you expand the build step and scroll to the end, you can see the errors. No idea why the overall CI says success.

Reply:
hmmm, maybe instead of updating the gcc version for every codecov ci, we should update our builder

https://github.com/awslabs/aws-crt-builder/blob/42d15ba4ee3fbe0e812e25094a8cf59eaef0cf1e/builder/actions/cmake.py#L254-L258

We install our own version of gcc, but the `ctest -T coverage` probably still use the default one in the system.

Seems like we can pass `--gcov` to chose what version of gcov to solve the version mismatch thing. https://stackoverflow.com/questions/12454175/gcov-out-of-memory-mismatched-version

Reply:
Thanks, fixed by https://github.com/awslabs/aws-crt-builder/pull/317.